RetailCoder.VBE/UnitTesting/PermissiveObjectComparer.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Rubberduck.UnitTesting
+{
+    internal class PermissiveObjectComparer : IEqualityComparer<object>
+    {
+        /// <summary>
+        /// Tests equity between 2 objects using VBA's type promotion rules.
+        /// </summary>
+        /// <param name="x"></param>
+        /// <param name="y"></param>
+        /// <returns>VBA equity</returns>
+        public new bool Equals(object x, object y)
+        {
+            var expected = x;
+            var actual = y;
+
+            // try promoting integral types first.
```

Comment:
I'm pretty sure that the `ulong` cast would catch all of the non-floating point types from `LongLong` down to `Byte`.